tests/unit/components/cylc/toolbar.vue.spec.js

```diff
@@ -1,55 +1,53 @@
-import { mount } from '@vue/test-utils'
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
 import { expect } from 'chai'
-// import vuetify here so that we do not have warnings in the console output
-// eslint-disable-next-line no-unused-vars
-import * as vuetify from '@/plugins/vuetify'
 import Toolbar from '@/components/cylc/Toolbar'
 import TaskState from '@/model/TaskState.model'
 import store from '@/store/index'
 import Vue from 'vue'
+import Vuetify from 'vuetify'
+
+const localVue = createLocalVue()
 
 describe('Toolbar component', () => {
+  let vuetify
+  beforeEach(() => {
+    vuetify = new Vuetify()
+  })
   it('should initialize props', () => {
-    const wrapper = mount(Toolbar, {
-      store,
-      propsData: {
-        workflow: {
-          id: 'Workflow Test Toolbar',
-          status: TaskState.RUNNING.name.toLowerCase()
-        }
-      }
+    const wrapper = shallowMount(Toolbar, {
+      store
     })
     expect(wrapper.is(Toolbar)).to.equal(true)
   })
-  it('should hide and display drawer according to screen viewport size', () => {
+  it('should hide and display drawer according to screen viewport size', async () => {
+    store.state.workflows.workflow = {
+      id: 'Workflow Test Toolbar',
+      status: TaskState.RUNNING.name.toLowerCase()
+    }
+    // v-app-toolbar when using "app" directive, must also have a v-app. So we need to initialize vuetify
     const wrapper = mount(Toolbar, {
-      store,
-      propsData: {
-        workflow: {
-          id: 'Workflow Test Toolbar',
-          status: TaskState.RUNNING.name.toLowerCase()
-        }
-      }
+      localVue,
```

Comment:
What does this do?

Reply:
It is from vue-test-utils, and allows us to create a local vue instance that does not pollute global vue instance.

This allows multiple tests to hack the attributes of Vue without breaking other tests (changes are scoped to the test, function, closure, etc, where you created it).

In this case, we are hacking the local vue to add vuetify, and tests of other components should not be affected by that (at least that's the theory).

Reply:
Thanks for the info.